libcxx/include/__locale

```diff
@@ -372,7 +372,7 @@ public:
 #else
     static const mask __regex_word = 1<<10;
 #endif // defined(__BIONIC__)
-#elif defined(__GLIBC__)
+#elif defined(__GLIBC__) || defined(__BIG_ENDIAN__)
```

Comment:
It's strange to me that nothing else mentions endian-ness here. Is there a different macro we could also switch on?

Reply:
I found the relevant changes:[D17132](https://reviews.llvm.org/D17132l), In this changes, the value of __regex_word is determined by whether the GLIBC is defined to determine whether the value is big-endian.

Reply:
Sorry, there's something wrong with this modification. I've re-initiated a pr.